pylint/checkers/refactoring.py

```diff
@@ -311,6 +317,38 @@ def leave_functiondef(self, _):
         # new scope = reinitialize the stack of nested blocks
         self._nested_blocks = []
 
+    def visit_raise(self, node):
+        self._check_stop_iteration_inside_generator(node)
+
+    def _check_stop_iteration_inside_generator(self, node):
+        """
+        Check if an exception of type StopIteration is raised inside a generator
+        """
+        scope = node.scope()
+        if isinstance(scope, astroid.FunctionDef) and scope.is_generator():
```

Comment:
maybe invert if condition and do early return instead? It'll help with deep indentation level.